src/views/credentials/__tests__/viewCredentialsList.test.tsx

```diff
@@ -0,0 +1,294 @@
+import { CredentialType } from '../../../types/types';
+
+describe('viewCredentialsList', () => {
+  const mockCredential: CredentialType = {
+    id: 1,
+    name: 'Test Credential',
+    cred_type: 'network',
+    sources: [],
+    username: 'test',
+    password: '',
+    ssh_key: '',
+    auth_token: '',
+    ssh_passphrase: '',
+    become_method: '',
+    become_user: '',
+    become_password: '',
+    auth_type: 'username_password',
+    created_at: new Date(),
+    updated_at: new Date()
+  };
+
+  describe('Checkbox selection state', () => {
+    it('should handle single item selection', () => {
+      const selectedItems = [mockCredential];
+      const isSelected = (item: CredentialType) => {
+        if (Array.isArray(selectedItems)) {
+          return selectedItems.some(selected => selected.id === item.id);
+        }
+        return !!selectedItems[item.id];
+      };
+
+      expect(isSelected(mockCredential)).toBe(true);
+      expect(isSelected({ ...mockCredential, id: 999 })).toBe(false);
+    });
```

Comment:
Can you help me understand what does it test?

It looks like in this test an array is created, a function is defined and then there are assertions this function behaves a certain way when running on that array or another. But I don't see how it exercises what we have in our codebase.

In fact, I can go as far as check out this branch, remove `src/views/credentials/viewCredentialsList.tsx` file and `npx jest ./src/views/credentials/__tests__/viewCredentialsList.test.tsx` will still pass.